### PR TITLE
Refactor id tags

### PR DESCRIPTION
**Pull request by Ozaq**
So far is have been modelled with 'identifiable', the implementation was
changed to allow copy of ids and to prefer composition over inheritance.

The type 'Identifiable' has been renamed to 'UniqueId' to represent that
change, further the type has become formattable and can no longer be
converted to an integer. The idea here is that you should create
interfaces that rely on a specific id type for the "thing" you want to
identify. This way you cannot mix ids of different domains.

If there are external IDs that track the progress of, for example an
agent, you need to maintain a mapping from your external id to the id
generated by jpscore.

As part of this refactoring 'special_area' lost its external_id property
since this is not part of the domain of jpscore.

### CODE REVIEW

File: cpp/libcore/source/util/identifiable.hpp [hunk 1]
```diff
@@ -1,40 +1,78 @@
 #pragma once
```
Comment (zeroset) [inline cpp/libcore/source/util/identifiable.hpp:1]: Can you please rename the file to `unique_id.hpp`

File: cpp/libcore/source/util/identifiable.hpp [hunk 2]
```diff
@@ -1,40 +1,78 @@
...
+class UniqueId
 {
     /// Static count for the next available unique identifier.
     // As there is a bug in clang-tidy (https://bugs.llvm.org/show_bug.cgi?id=48040) the nolint is
     // needed for now NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
-    inline static std::atomic<unsigned int> uid_counter{0};
+    inline static std::atomic<Integer> uid_counter{0};
     /// The actual unique identifier of the object
-    unsigned int m_id{uid_counter++};
+    Integer m_value{++uid_counter};
```
Comment (zeroset) [inline cpp/libcore/source/util/identifiable.hpp:26]: Why did you change to pre increment here?

File: cpp/libcore/test/util/test_identifiable.cpp [hunk 3]
```diff
@@ -1,50 +1,41 @@
 #include "util/identifiable.hpp"
```
Comment (zeroset) [inline cpp/libcore/test/util/test_identifiable.cpp:1]: Please rename the file to `test_unique_id.cpp`

**Review by zeroset — CHANGES_REQUESTED** [review]

File: cpp/libcore/source/util/identifiable.hpp (hunk 1 above)
Comment (Ozaq) [reply cpp/libcore/source/util/identifiable.hpp]: done

File: cpp/libcore/test/util/test_identifiable.cpp (hunk 3 above)
Comment (Ozaq) [reply cpp/libcore/test/util/test_identifiable.cpp]: done

File: cpp/libcore/source/util/identifiable.hpp (hunk 2 above)
Comment (Ozaq) [reply cpp/libcore/source/util/identifiable.hpp]: To keep `0` out of the valid (internal) ids. To have the possibility to express an invalid id, e.g. for testing

**Review by zeroset — APPROVED** [review]

**Pull request merged**